src/docs/rts-new.md

```diff
@@ -0,0 +1,178 @@
+# Interaction with timestamps
+
+The main timestamps RTS are concerned with are the durable and stable
+timestamps. The high-level details are covered above, what follows is a more
+detailed view of how we use these timestamps.
+
+In order to ensure that the system is in a quiescent state, in diagnostic
+builds we read the global pinned and stable timestamps at the start and end
+of RTS to assert they didn't change underneath us.
+
+Once RTS has finished, we roll back the global durable timestamp to the
+stable timestamp, since if there are no unstable updates, the global durable
+timestamp can, by definition, be moved backwards to the stable timestamp.
+
+Timestamps of updates in the history store can have some non-obvious
+properties:
+
+* The start timestamp can be equal to the stop timestamp if the original
+  update's commit timestamp is in order.
+* During rollback of a prepared transaction, a history store update with a
+  stop timestamp may not get removed, leading to a duplicate record. RTS
+  ignores these.
+* If we've had to fix any timestamps as part of RTS, a newly inserted update
+  with an older timestamp may have a durable timestamp smaller than the
+  current stop durable timestamp.
+* When a checkpoint writes a committed prepared update, it's possible for
+  further updates on that key to include history store changes before the
+  transaction alters the history store update to have a proper stop
+  timestamp. Thus, we can have an update in the history store with a maximum
+  stop timestamp in the middle of other updates for the same key.
+  
+We also have some verification specifically related to the history
+store. While iterating over records, we check that the start time of a newly
+encountered record is greater than or equal to the previous record. We also
+validate that the timestamps in the key and the cell are the same (where
+possible).
+
+Fast-truncate is worth mentioning briefly. If the page has a page delete
+structure, we can look at its durable timestamp, much the same as we would a
+normal page.
+
+# Interaction with transaction IDs
+
+This is a little simpler than the timestamp rules - we use normal snapshot
+visibility in most of the places we look at timestamps (see the timestamps
+section).
+
+We also clear the transaction ID of updates we look at during RTS, because
+the connection's write generation will be initialised after RTS and the
```

Comment:
Use American spelling: `initialized`

Reply:
Normally I say thanks or something, but for this one I'm just going to passive-aggressively link to the fix ;-) 5ec1700

Reply:
Well, all I can say is that when Michael and Keith started WT they made a conscious decision to standardize on that.